interfaces/types.go

```diff
@@ -0,0 +1,8 @@
+package interfaces
+
+// CustomTypePaginator is an interface that custom types need to implement
+// in order to allow pagination over fields inside custom types.
+type CustomTypePaginator interface {
```

Comment:
I personally would prefer to call it `CustomType`, just my preference though 😆

And do you think it's reasonable to put this interface into `cursor` package? I see now only `cursor` package would depend on this interface.

Reply:
`CustomType` also sounds good to me. Renamed to that.

Also, moved it to `cursor`. I thought that introduced some cycle import issues before, so I think this was the reason to move it into a separate package. But it seem OK now in the `cursor` packer so let's have it there.